renku/data/pre-commit.sh

```diff
@@ -40,6 +40,12 @@ fi
 
```

Comment:
It'd make sense to add explicit commands that check if a project is newer or outdated and check for it here, after the check if renku is installed. Something like `renku project-status` or `renku migrate --check` that returns a message regarding the project state and an exit code that goes with it (like 2= migration required, 3= project too new).
Then we can check for the exit code and write a message "Project metadata is outdated, migrate it with `renku migrate`" or "This project was created with a newer version of renku, update renku to continue working on it" or something to that effect.

Just to make it more explicit to the user what is happening and what steps they need to take to solve the problem.

Reply:
I've separated outdated and unsupported cases and also added `renku migrate --check`. However, I did not use it in the pre-hook because running another renku command there would slow-down commits.